game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/MainDefenseCombatValue.java

```diff
@@ -0,0 +1,138 @@
+  public StrengthAndRollCalculator getRoll() {
+    return new MainDefenseRoll(supportFromFriends, supportFromEnemies);
+  }
+
+  @Override
+  public StrengthAndRollCalculator getStrength() {
+    return new MainDefenseStrength(
+        gameData, supportFromFriends, supportFromEnemies, territoryEffects);
+  }
+
+  @Override
+  public boolean isDefending() {
+    return true;
+  }
+
+  static class MainDefenseRoll extends StrengthAndRollCalculator {
+
+    MainDefenseRoll(final AvailableSupports friendlySupport, final AvailableSupports enemySupport) {
+      super(friendlySupport, enemySupport);
+    }
+
+    @Override
+    public int getValue(final Unit unit) {
+      return RollValue.of(unit.getUnitAttachment().getDefenseRolls(unit.getOwner()))
+          .add(addSupport(unit, friendlySupportTracker))
+          .add(addSupport(unit, enemySupportTracker))
+          .getValue();
+    }
+
+    @Override
+    protected Predicate<UnitSupportAttachment> getRuleFilter() {
+      return UnitSupportAttachment::getRoll;
+    }
+  }
+
+  static class MainDefenseStrength extends StrengthAndRollCalculator {
+
+    private final GameData gameData;
+    private final Collection<TerritoryEffect> territoryEffects;
+
+    MainDefenseStrength(
+        final GameData gameData,
+        final AvailableSupports friendlySupport,
+        final AvailableSupports enemySupport,
+        final Collection<TerritoryEffect> territoryEffects) {
+      super(friendlySupport, enemySupport);
+      this.gameData = gameData;
+      this.territoryEffects = territoryEffects;
+    }
+
+    @Override
+    public int getValue(final Unit unit) {
+      int strength = unit.getUnitAttachment().getDefense(unit.getOwner());
+      boolean allowFriendly = true;
+      if (isFirstTurnLimitedRoll(unit.getOwner())) {
+        // if first turn is limited, the strength is a max of 1 and no friendly support
+        strength = Math.min(1, strength);
+        allowFriendly = false;
+      }
+      StrengthValue strengthValue =
+          StrengthValue.of(gameData.getDiceSides(), strength)
+              .add(
+                  TerritoryEffectHelper.getTerritoryCombatBonus(
+                      unit.getType(), territoryEffects, true));
+
+      if (allowFriendly) {
+        strengthValue = strengthValue.add(addSupport(unit, friendlySupportTracker));
+      }
+      strengthValue = strengthValue.add(addSupport(unit, enemySupportTracker));
+      return strengthValue.getValue();
+    }
+
+    private boolean isFirstTurnLimitedRoll(final GamePlayer player) {
+      // If player is null, Round > 1, or player has negate rule set: return false
+      return !player.isNull()
+          && gameData.getSequence().getRound() == 1
+          && !isNegateDominatingFirstRoundAttack(player)
+          && isDominatingFirstRoundAttack(gameData.getSequence().getStep().getPlayerId());
+    }
+
+    private boolean isNegateDominatingFirstRoundAttack(final GamePlayer player) {
+      final RulesAttachment ra =
```

Comment:
nit, please prefer to spell out variable names.

`ra => rulesAttachment`

Reply:
Ok.  This was moved code so I didn't make any changes to it.